url-updater/src/main/java/com/devonfw/tools/ide/url/folderhandling/UrlChecksum.java

```
@@ -0,0 +1,129 @@
+        super(parent, name + ".sha256");
+    }
+
+    public String getChecksum() {
+        return checksum;
+    }
+
+    public void setChecksum(String checksum) {
+        this.checksum = checksum;
+    }
+
+    /**
+     * @param url the url of the download file
+     * @return the input stream of requested url
+     */
+    private InputStream doGetResponseInputStream(String url) {
+        try {
+            HttpRequest request1 = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .GET()
+                    .build();
+            return client.send(request1, HttpResponse.BodyHandlers.ofInputStream()).body();
+        } catch (IOException | InterruptedException exception) {
+            throw new IllegalStateException("Failed to retrieve response body from url: " + url, exception);
+        } catch (IllegalArgumentException e) {
+            System.out.println("Error while getting response body from url {}\", url, e");
+            return null;
+        }
+    }
+
+    /**
+     * @param inputStream the input stream of requested url
+     * @return checksum of input stream as string
+     */
+    private String doGenerateChecksumFromInputStream(InputStream inputStream)  {
+
+        try {
+            MessageDigest md = MessageDigest.getInstance(hashAlgorithm);
+
+            byte[] buffer = new byte[8192];
+            int bytesRead;
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                md.update(buffer, 0, bytesRead);
+            }
+            inputStream.close();
+
+            byte[] digestBytes = md.digest();
+            String checksum = toHexString(digestBytes);
+            System.out.println(checksum);
+            return checksum;
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to read input stream " + e);
+        } catch (NoSuchAlgorithmException e) {
+            System.out.println("No such hash algorithm " + hashAlgorithm);
```

Also throw an `IllegalStateException` instead of returning `null`. In this case the JDK is broken, and we should react and fix the problem rather than trying to continue...

---

done